### PR TITLE
[Fix] GET Request are failing due to variablesKey cast that always fails.

### PR DESCRIPTION
**Pull request by dmandarino**
#623

GET request are always failing because of a cast for in [String: AnyHashable] `GraphQLGETTransformer`. So I'm changing it to `[AnyHashable : Any]` because I've checked that it's the type we need to use to use custom headers too.

### CODE REVIEW

**Comment by dmandarino** [comment]
@designatednerd Sure, I'm taking a look on it.

**Comment by dmandarino** [comment]
@designatednerd it's still stucked. Is it usual to take that long? The other time it was so fast.

**Comment by designatednerd** [comment]
Yeah during the day in the US the wait time for builds spikes pretty significantly, unfortunately. 😭

**Comment by designatednerd** [comment]
@dmandarino looks like there's a couple failures in there where occasionally the order of parameters is not the same. I'll take a longer look tomorrow but I suspect the dictionary you're encoding is being generated in a different order. 

I'm annoyed that [`.sortedKeys` for JSONSerialization](https://developer.apple.com/documentation/foundation/jsonserialization/writingoptions/2888322-sortedkeys) is iOS 11+, I might have to write something for that to support it on older devices.

File: Sources/Apollo/GraphQLGETTransformer.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ struct GraphQLGETTransformer {
     queryItems.append(URLQueryItem(name: self.queryKey, value: query))
     components.queryItems = queryItems
     
-    guard let variables = self.body.jsonObject[self.variablesKey] as? [String: AnyHashable] else {
+    guard let variables = self.body.jsonObject[self.variablesKey] as? [AnyHashable : Any] else {
```
Comment (designatednerd) [inline Sources/Apollo/GraphQLGETTransformer.swift]: So messing around in a playground then looking at [the docs for `JSONSerialization`](https://developer.apple.com/documentation/foundation/jsonserialization) I think this needs to be `[String: Any]` - JSON Serialization barfs if the keys aren't strings.

**Comment by designatednerd** [comment]
Ugh, so I looked into this and getting `JSONSerialization` to sort consistently without using `.sortedKeys` is *stupidly* difficult. 

I'm almost certainly going to move codegen stuff to `Codable` in the medium term, so I didn't want to go too far down the rabbit hole of trying to get this to work with `JSONSerialization`, but it made the test about 900% more annoying to write. 

I've opened #628 with my solution to this - do you mind if we move forward with that one since I've already dealt with the test nonsense? Your work here was absolutely critical to getting that to work, and I really appreciate it - I'll make sure this PR gets linked in the changelog!

**Comment by dmandarino** [comment]
@designatednerd don't worry, just do it! I just wanted to help this project. We all need this feature, specially for persisted query. And I really need this GET, because right now I'm using a fork of my own so the company I'm working on can use Apollo Client for iOS.

**Comment by designatednerd** [comment]
I don't mind at all - let me know what you think, and I'll close this one out.

**Pull request closed without merging**